src/hooks/useConditionalHandler.ts

```diff
@@ -0,0 +1,20 @@
+import { useCallback } from 'react'
+
+export function useConditionalHandler<T extends (...args: any[]) => void | boolean | Promise<boolean>>(
+  handler?: T
+): (...params: Parameters<T>) => Promise<boolean> {
+  return useCallback(
+    async (...params) => {
+      if (!handler) return true
+
+      try {
+        const result = await handler(...params)
+        if (result === false) return false
+      } catch {
+        return false
+      }
+      return true
```

Comment:
can replace these returns with a single `finally` block

Reply:
I think a `finally` block will always run, so it overrides the returned value from within the try-catch block.